Data/Pool.hs

```diff
@@ -210,6 +214,27 @@ createPool' create destroy numStripes idleTime maxResources timeout = do
     V.mapM_ (\lp -> mkWeakIORef (lfin lp) (purgeLocalPool destroy lp)) localPools
   return p
 
+resizePool :: Pool a -> Int -> IO ()
+resizePool Pool{..} maxResources' = do
+  when (maxResources' < 1) $
+    modError "pool " $ "invalid maximum resource count " ++ show maxResources'
+  atomically $ writeTVar maxResources maxResources'
+
+tryTrimLocalPool :: (a -> IO ()) -> TVar Int -> LocalPool a -> IO ()
+tryTrimLocalPool destroy maxResources LocalPool{..} = do
+  resources <- atomically $ do
+    maxSize <- readTVar maxResources
+    curSize <- readTVar inUse
+    let desiredDrop = curSize - maxSize
```

Comment:
I'm assuming here `desiredDrop` is expected to be a positive integer. So, my question is how is `curSize` >= `maxSize`?

Reply:
This can happen when the pool's maximum size was recently reduced.

- `curSize > maxSize`: we have acquired too many connections, and should try to drop a few. `tryTrimLocalPool` tries to drop any unused connections, up until `curSize = maxSize`. it won't manage if too many connections are in use.
- `curSize = maxSize`: we're at our connection allocation maximum - no action required
- `curSize < maxSize`: we still have space to grow - no action required. in this case `desiredDrop` will be negative, and `splitAt desiredDrop xs = ([], xs)`.

Reply:
Oh, you have already added it! Thanks!